spec/jobs/problem_check_spec.rb

```diff
@@ -56,8 +61,24 @@
     expect(problems.map(&:to_s)).to match_array(["yuge problem"])
   end
 
+  it "schedules a retry if there are attempts remaining" do
+    AdminDashboardData.add_scheduled_problem_check(:test_identifier, TestCheck) do
+      AdminDashboardData::Problem.new("big problem")
+    end
+
+    TestCheck.stubs(:retries).returns(1)
+
+    expect_enqueued_with(
+      job: :problem_check,
+      args: {
+        check_identifier: :test_identifier,
+        retry_count: 1,
```

Comment:
Making sure we increment by one here so we don't end up endlessly scheduling jobs with `retry_count: 0`. 😬

Reply:
Maybe we should also validate that `retry_count` is not `1` or `0` in the job?

Reply:
Actually I kind of feel like going further and check that it doesn't schedule a retry when the `max_retry` is reached.